app/controllers/tasks_controller.rb

```diff
@@ -0,0 +1,82 @@
+class TasksController < ApplicationController
+  def index
+    @tasks = Task.all
+  end
+
+  def show
+    task_id = params[:id].to_i
+    @task = Task.find_by(id: task_id)
+
+    if @task.nil?
+      redirect_to tasks_path and return
+    end
+  end
+
+  def new
+    @task = Task.new
+  end
+
+  def create
+    @task = Task.new(task_params)
+
+    if @task.save
+      redirect_to task_path(@task) and return
+    else
+      render :new and return
```

Comment:
It's helpful to return a non-success status when you render after failing to save:

```suggestion
      render :new, status: :bad_request and return
```